logger/logger.go

```diff
@@ -182,7 +182,20 @@ func convertLogLevel(level string) zerolog.Level {
 }
 
 func setupCloudwatchLogging(conf *CloudWatchConfiguration) (io.Writer, error) {
-	conf.StreamName = strings.ReplaceAll(conf.StreamName, "$HOSTNAME", os.Getenv("HOSTNAME"))
+	// os.Hostname is preferred to os.Getenv("HOSTNAME") because the env var might
+	// not be populated yet (e.g. GitHub runners)
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	// if no log stream name is explicitly provided, HOSTNAME is used
+	if conf.StreamName == "" {
+		conf.StreamName = hostname
+	} else {
+		// take provided log stream name and replace any $HOSTNAME placeholders with real hostname
+		conf.StreamName = strings.ReplaceAll(conf.StreamName, "$HOSTNAME", hostname)
```

Comment:
in which case would this happen?

Reply:
I'm not sure, but I'm keeping the original functionality in order to not break something accidentally.